<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>929</width>
    <height>631</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background: orange</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QSplitter" name="splitter">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="FormAnimTimelineWithSession" name="widget_timeline" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
       <widget class="QWidget" name="widget" native="true"/>
      </widget>
      <widget class="QGroupBox" name="groupBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Input/Output</string>
       </property>
       <widget class="QPushButton" name="pushButton_clearKeyPoses">
        <property name="geometry">
         <rect>
          <x>450</x>
          <y>80</y>
          <width>101</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>clearKeyPoses</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="doubleSpinBox_startOut">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>160</y>
          <width>62</width>
          <height>22</height>
         </rect>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="maximum">
         <double>9999.000000000000000</double>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="cursorOut">
        <property name="geometry">
         <rect>
          <x>310</x>
          <y>160</y>
          <width>71</width>
          <height>21</height>
         </rect>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="minimum">
         <double>-100.000000000000000</double>
        </property>
        <property name="maximum">
         <double>9999.000000000000000</double>
        </property>
       </widget>
       <widget class="QPushButton" name="end">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>140</y>
          <width>80</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>end</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="doubleSpinBox_endIn">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>120</y>
          <width>62</width>
          <height>22</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="durationOut">
        <property name="geometry">
         <rect>
          <x>750</x>
          <y>160</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="maximum">
         <double>9999.000000000000000</double>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="durationIn">
        <property name="geometry">
         <rect>
          <x>750</x>
          <y>119</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="prefix">
         <string/>
        </property>
       </widget>
       <widget class="QPushButton" name="start">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>140</y>
          <width>80</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>start</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_showTimeline">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>60</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>show timeline</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="doubleSpinBox_startIn">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>120</y>
          <width>62</width>
          <height>22</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="pause">
        <property name="geometry">
         <rect>
          <x>580</x>
          <y>160</y>
          <width>80</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Pause</string>
        </property>
       </widget>
       <widget class="QPushButton" name="addKeyPose">
        <property name="geometry">
         <rect>
          <x>450</x>
          <y>140</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>add keyPose</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="keyPoseOut">
        <property name="geometry">
         <rect>
          <x>450</x>
          <y>160</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="maximum">
         <double>9999.000000000000000</double>
        </property>
       </widget>
       <widget class="QPushButton" name="changeDuration">
        <property name="geometry">
         <rect>
          <x>750</x>
          <y>140</y>
          <width>101</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>duration</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="doubleSpinBox_endOut">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>160</y>
          <width>62</width>
          <height>22</height>
         </rect>
        </property>
        <property name="decimals">
         <number>4</number>
        </property>
        <property name="maximum">
         <double>9999.000000000000000</double>
        </property>
       </widget>
       <widget class="QPushButton" name="play">
        <property name="geometry">
         <rect>
          <x>580</x>
          <y>130</y>
          <width>80</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Play</string>
        </property>
       </widget>
       <widget class="QPushButton" name="changeCursor">
        <property name="geometry">
         <rect>
          <x>310</x>
          <y>140</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>cursor</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="nbKeyPose">
        <property name="geometry">
         <rect>
          <x>450</x>
          <y>120</y>
          <width>43</width>
          <height>22</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="cursorIn">
        <property name="geometry">
         <rect>
          <x>310</x>
          <y>120</y>
          <width>71</width>
          <height>21</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="prefix">
         <string/>
        </property>
        <property name="suffix">
         <string/>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_hideTimeline">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>60</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>hide timeline</string>
        </property>
       </widget>
       <widget class="QSlider" name="anim">
        <property name="geometry">
         <rect>
          <x>390</x>
          <y>240</y>
          <width>160</width>
          <height>16</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>330</x>
          <y>240</y>
          <width>52</width>
          <height>14</height>
         </rect>
        </property>
        <property name="text">
         <string>anim</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>929</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FormAnimTimelineWithSession</class>
   <extends>QWidget</extends>
   <header>AnimTimeline/FormAnimTimeline.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <slot>cursorChanged(double)</slot>
  <slot>onChangeCursor(double)</slot>
  <slot>onPlay()</slot>
  <slot>onPause()</slot>
  <slot>onAddKeyPose(double)</slot>
  <slot>onKeyPoseChanged(int)</slot>
  <slot>onRemoveKeyPose(int)</slot>
 </slots>
</ui>
